modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -49,21 +49,51 @@ def submit_supporting_documents
 
       private
 
-      def handle_file_uploads(form_id, parsed_form_data)
-        file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
-        statuses, error_message = FileUploader.new(form_id, metadata, file_paths, true).handle_uploads
-        statuses = Array(statuses)
-
-        # Retry attempt if specific error message is found
-        if statuses.any? do |status|
-          status.is_a?(String) && status.include?('No such file or directory @ rb_sysopen')
+      # rubocop:disable Layout/LineLength
+      if Flipper.enabled?(:champva_multiple_stamp_retry, @user)
+        def handle_file_uploads(form_id, parsed_form_data)
+          attempt = 0
+          max_attempts = 1 # Only one retry attempt
+
+          begin
+            file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
+            file_uploader = FileUploader.new(form_id, metadata, file_paths, true)
+            statuses, error_message = file_uploader.handle_uploads
+          rescue => e
+            attempt += 1
+            error_message_downcase = e.message.downcase
+            Rails.logger.error "Error handling file uploads (attempt #{attempt}): #{e.message}"
+
+            if (error_message_downcase.include?('failed to generate stamped file') || error_message_downcase.include?('unable to find file')) && attempt <= max_attempts
+              Rails.logger.error 'Retrying in 2 seconds...'
+              sleep 2
+              retry
+            else
+              return [[], 'Error handling file uploads']
+            end
+          end
+
+          [statuses, error_message]
         end
+
+      else
+        def handle_file_uploads(form_id, parsed_form_data)
           file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
           statuses, error_message = FileUploader.new(form_id, metadata, file_paths, true).handle_uploads
-        end
+          statuses = Array(statuses)
 
-        [statuses, error_message]
+          # Retry attempt if specific error message is found
+          if statuses.any? do |status|
+            status.is_a?(String) && status.include?('No such file or directory @ rb_sysopen')
+          end
+            file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
+            statuses, error_message = FileUploader.new(form_id, metadata, file_paths, true).handle_uploads
```

Comment:
This line is where the retry happens.  If you comment it out, the unit test that checks for the retry doesn't fail.